### ISSUE TITLE
False positives for react/display-name in HOF returning null

### ISSUE THREAD

**Issue by jxm-math** [jsx-eslint/eslint-plugin-react#3346]
Any of these constructs are incorrectly identified as Components:

```javascript
any = () => () => null
{ any: () => () => null }
any = function() {return function() {return null;};}
{ any: function() {return function() {return null;};} }
```
